### PR TITLE
Feat: Add useTheme() and makeStyles() helpers

### PR DESCRIPTION
**Pull request by MarianAlexandruAlecu**
I've been using material-ui for web for a few years now and I've observed that two useful helpers (`useTheme()` and `makeStyles()`) are missing from this repo.

`useTheme()` is just a shorthand for `useContext(ThemeContext)`

`makeStyles()` enables a simpler way of using the theme in styles:

Instead of passing the theme from the component to `createStyles()`
```
const Component = (props: Props) => {
    const { theme } = useContext(ThemeContext);
    const styles = createStyles(theme, props);
    ....
}

const createStyles = (theme: Partial<FullTheme>, props) => ({
    container: {
          color: theme.colours?.primary
     }
});
```

we can just do access it directly in `makeStyles()`
```
const Component = (props: Props) => {
     const styles = useStyles(props);
     ....
}

const useStyles = makeStyles((theme, props: Props) => ({
    container: {
          color: theme.colours?.primary
     }
}));
```

Bonus: make transition from React web/material-ui to React Native/react-native-elements easier 🚀

### CODE REVIEW

**Review by flyingcircle — CHANGES_REQUESTED** [review]
I like the additions but 2 things:
1. There is no need to update the package version. That is done whenever a release is determined to happen. So no need to include that in the PR.
2. I just added a huge change converting the codebase to TS. Please pull that in to your branch and rework the code. Hopefully it isn't too much of a burden.

Thanks

**Comment by MarianAlexandruAlecu** [comment]
> I like the additions but 2 things:
> 
> 1. There is no need to update the package version. That is done whenever a release is determined to happen. So no need to include that in the PR.
> 2. I just added a huge change converting the codebase to TS. Please pull that in to your branch and rework the code. Hopefully it isn't too much of a burden.
> 
> Thanks

Thanks for feedback, I've updated the PR 🤓

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file. Dates are d
 
 Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 
+#### [v2.3.3]
```
Comment (flyingcircle) [inline CHANGELOG.md:7]: This is not the correct version that this change would be in. You can remove this section. I'll add it to the release notes of the version that this ends up making it into.

**Review by flyingcircle — CHANGES_REQUESTED** [review]
Great! Thanks for the update to typescript. I left one comment in the changelog and the only other thing is to add a small section showing the API with an example in `website/docs/customiztion.md` so that people can learn about the new API functionality.

**Comment by flyingcircle** [comment]
Don't give up on me now @MarianAlexandruAlecu! I'm hoping to do a release soon. Are you going to be able to get the documentation part written or is it going to be a few days until you can finish out the documentation?

**Comment by MarianAlexandruAlecu** [comment]
> Don't give up on me now @MarianAlexandruAlecu! I'm hoping to do a release soon. Are you going to be able to get the documentation part written or is it going to be a few days until you can finish out the documentation?
> 
> Edit: and unit tests

On it now 🤓

**Comment by flyingcircle** [comment]
I think you might be able to fix the "conflicting files" issue by renaming `src/index.tsx` to `src/index.js`. Hopefully that should resolve that issue.

**Comment by MarianAlexandruAlecu** [comment]
Ready 🚀 

Btw, shouldn't `src/index.js` be a TS file? 🤔

**Comment by flyingcircle** [comment]
I guess it could be, but it just re-exports stuff so there's nothing typescripty in it either way. Typescript still creates an `index.d.ts` file, so it doesn't break anything. Thanks for the quick turn around on this. I'll try to check it out and test it later today.

**Review by flyingcircle — APPROVED** [review]

**Comment by MarianAlexandruAlecu** [comment]
> I guess it could be, but it just re-exports stuff so there's nothing typescripty in it either way. Typescript still creates an `index.d.ts` file, so it doesn't break anything. Thanks for the quick turn around on this. I'll try to check it out and test it later today.

Totally agree, it's just consistency in general: if a project is TS, you usually put .ts or .tsx everywhere without thinking if it's necessary or not.

**Comment by flyingcircle** [comment]
@MarianAlexandruAlecu Thanks much! I usually like to say thank you for submissions to the repo with a token $25. Feel free to contact me on the RNE slack for details if you're interested.

**Comment by MarianAlexandruAlecu** [comment]
I'm grateful that I was able to write even a small piece for a library that used!  Cheers! 🤝

**Pull request merged**